rasa/core/policies/policy.py

```diff
@@ -120,12 +126,17 @@ def __init__(
         self.__featurizer = self._create_featurizer(featurizer)
         self.priority = priority
         self.finetune_mode = should_finetune
+        self._rule_only_data = {}
 
     @property
     def featurizer(self):
         """Returns the policy's featurizer."""
         return self.__featurizer
 
+    def set_rule_only_data(self, rule_only_data: Dict[Text, Any]) -> None:
```

Comment:
how about using something which generalizes better?

```python
def set_shared_policy_state(**kwargs: Any) -> None
```

Advantages:
* no deprecation is required if we want to pass more information / different information
* users can use this for their own policies (e.g. setting some shared state originating from their policies)

Reply:
users wouldn't be able to use it, since ensemble will only pass `rule_only_data`

Reply:
renamed the method